test/jdk/jdk/internal/platform/cgroup/TestCgroupSubsystemFactory.java

```diff
@@ -190,18 +228,22 @@ public void teardown() {
     public void testCgroupv1JoinControllerCombo() throws IOException {
         String cgroups = cgroupv1CgroupsJoinControllers.toString();
         String mountInfo = cgroupv1MountInfoJoinControllers.toString();
-        Optional<CgroupTypeResult> result = CgroupSubsystemFactory.determineType(mountInfo, cgroups);
+        String selfCgroup = cgroupv1SelfCgroupJoinCtrl.toString();
+        Optional<CgroupTypeResult> result = CgroupSubsystemFactory.determineType(mountInfo, cgroups, selfCgroup);
 
         assertTrue("Expected non-empty cgroup result", result.isPresent());
         CgroupTypeResult res = result.get();
         assertFalse("Join controller combination expected as cgroups v1", res.isCgroupV2());
+        CgroupInfo memoryInfo = res.getInfos().get("memory");
+        assertEquals("/user.slice/user-1000.slice/session-3.scope", memoryInfo.getCgroupPath());
```

Comment:
The gist of the Java equivalent of JDK-8253939 (which fixed this for Hotspot). I.e. a regression test for JDK-8217766 and JDK-8254854.